code/game/objects/items/weapons/implants/implant.dm

```diff
@@ -101,12 +101,36 @@
 
 /obj/item/weapon/implant/explosive/activate(var/cause)
 	if(!cause || !imp_in)	return 0
-	if(cause == "action_button" && alert(imp_in, "Are you sure you want to activate your explosive implant? This will cause you to explode and gib!", "Explosive Implant Confirmation", "Yes", "No") != "Yes")
+	if(cause == "action_button" && alert(imp_in, "Are you sure you want to activate your microbomb implant? This will cause you to explode!", "Microbomb Implant Confirmation", "Yes", "No") != "Yes")
 		return 0
-	explosion(src,0,1,5,7,10, flame_range = 5)
-	if(imp_in)
-		imp_in.gib()
+	var/light = 1
+	var/medium = 0.5
+	var/heavy = 0.25
+	for(var/obj/item/weapon/implant/explosive/E in imp_in)
+		heavy += 0.25
+		medium += 0.5
+		light += 1
+		if(E != src)
+			qdel(E)
+	heavy = round(heavy)
+	medium = round(medium)
+	light = round(light)
+	imp_in.gib()
+	explosion(src,0,heavy,medium,light,light, flame_range = medium)
+	qdel(src)
+
+/obj/item/weapon/implant/explosive/macro
+	name = "macrobomb implant"
+	desc = "And boom goes the weasel. And everything else nearby."
+	icon_state = "explosive"
 
+/obj/item/weapon/implant/explosive/macro/activate(var/cause)
+	if(!cause || !imp_in)	return 0
+	if(cause == "action_button" && alert(imp_in, "Are you sure you want to activate your macrobomb implant? This will cause you to explode and gib!", "Macrobomb Implant Confirmation", "Yes", "No") != "Yes")
+		return 0
+	imp_in.gib()
+	explosion(src,0,5,10,20,20, flame_range = 5)

```

Comment:
Same here, and if that's too much, at least up this to 10; 5 isn't going to impact anyone as they're gibbed or dead at that stage.